airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -81,7 +83,7 @@ public void start(final WorkerSourceConfig workerSourceConfig, final Path jobRoo
          */
         isResetBasedForConfig = false;
       } else {
-        stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState());
+        stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState(), useStreamCapableState);
```

Comment:
Just noting down a thought I had here: I was thinking this would break resets once we merge in the change to only reset new/updated streams on API update, since this would cause all states to deserialize into LEGACY, and the below logic would thrown an IllegalStateException in that case if not all streams are being reset.

But I think we should actually be okay because we should only ever be doing that if the persisted state of the connection is non-legacy, which will only happen when this feature flag is flipped to true. So I think we will be okay here.